pilz_control/test/unittest_pilz_joint_trajectory_controller.cpp

```diff
@@ -131,17 +151,57 @@ TEST_F(PilzJointTrajectoryControllerTest, testD0Destructor)
   SUCCEED();
 }
 
-TEST_F(PilzJointTrajectoryControllerTest, testInitializiation)
+/**
+ * @tests{Monitor_joint_accelerations,
+ *   Test if parameters for acceleration limits are correctly read.
+ * }
+ */
+TEST_F(PilzJointTrajectoryControllerTest, testGetJointAccelerationLimits)
 {
-  ASSERT_TRUE(manager_->loadController()) << "Failed to initialize the controller.";
+  // test setup
+  ros::NodeHandle nh{ "~" };
+  std::vector<std::string> joint_names;
+  ASSERT_TRUE(nh.getParam(CONTROLLER_JOINT_NAMES_PARAM, joint_names));
+  ASSERT_FALSE(joint_names.empty());
+
+  // test with existing acc limits
+  std::vector<boost::optional<double>> acceleration_limits = getJointAccelerationLimits(nh, joint_names);
+  EXPECT_EQ(joint_names.size(), acceleration_limits.size());
+  EXPECT_TRUE(acceleration_limits.at(0));
+  EXPECT_TRUE(acceleration_limits.at(1));
+  EXPECT_DOUBLE_EQ(*acceleration_limits.at(0), 3.49);
+  EXPECT_DOUBLE_EQ(*acceleration_limits.at(1), 3.49);  // as of pilz_control/test/config/joint_limits.yaml
+
+  // testing behaviour if `has_acceleration_limits` is false
```

Comment:
It seems to me that you test two objectives in one test. Maybe split into two tests?

Reply:
Done.